From 29dcbc5c25d6d8140337e96bf503c8475092c586 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Thu, 2 Dec 2010 16:14:27 -0800
Subject: [PATCH 1/2] xen: allocate irq descs on any NUMA node

Allocate irq descs on any NUMA node (we don't care) rather than
specifically node 0, which may not exist.

(At the moment NUMA is meaningless within a domain, so any info
the kernel has is just from an SRAT table we haven't suppressed/disabled.)

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
[ijc: backported to 2.6.32, effectively s/irq_alloc_desc_at/irq_to_desc_alloc_node/g]
---
 drivers/xen/events.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 2811bb9..0009e48 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -399,7 +399,7 @@
 	if (irq == start)
 		goto no_irqs;
 
-	desc = irq_to_desc_alloc_node(irq, 0);
+	desc = irq_to_desc_alloc_node(irq, -1);
 	if (WARN_ON(desc == NULL))
 		return -1;
 
@@ -592,7 +592,7 @@
 	 * we are using the !xen_initial_domain() to drop in the function.*/
 	if (identity_mapped_irq(gsi) || !xen_initial_domain()) {
 		irq = gsi;
-		irq_to_desc_alloc_node(irq, 0);
+		irq_to_desc_alloc_node(irq, -1);
 		dynamic_irq_init(irq);
 	} else
 		irq = find_unbound_irq();
